✨ [RUMF-1409] Provide setUser and related functions for logs SDK (#1801)
DataDog/browser-sdk: packages/logs/README.md

})

DD_LOGS.onReady(function () {
  DD_LOGS.getUser() // => {id: '1234', name: 'John Doe', email: 'john@doe.com', type: 'customer'}
})

DD_LOGS.onReady(function () {
  DD_LOGS.removeUserProperty('type')
})

DD_LOGS.onReady(function () {
  DD_LOGS.getUser() // => {id: '1234', name: 'John Doe', email: 'john@doe.com'}
})

DD_LOGS.onReady(function () {
  DD_LOGS.clearUser()
})

DD_LOGS.onReady(function () {
  DD_LOGS.getUser() // => {}
})
```

**Note:** Early API calls must be wrapped in the `DD_LOGS.onReady()` callback. This ensures the code only gets executed once the SDK is properly loaded.

##### CDN sync

For CDN sync, use:

```javascript
window.DD_LOGS && DD_LOGS.setUser({ id: '1234', name: 'John Doe', email: 'john@doe.com' })

window.DD_LOGS && DD_LOGS.setUserProperty('type', 'customer')

window.DD_LOGS && DD_LOGS.getUser() // => {id: '1234', name: 'John Doe', email: 'john@doe.com', type: 'customer'}

window.DD_LOGS && DD_LOGS.removeUserProperty('type')

window.DD_LOGS && DD_LOGS.getUser() // => {id: '1234', name: 'John Doe', email: 'john@doe.com'}

window.DD_LOGS && DD_LOGS.clearUser()

window.DD_LOGS && DD_LOGS.getUser() // => {}
```

> **Note**: The `window.DD_LOGS` check is used to prevent issues if a loading failure occurs with the SDK.

Reviewer: ```suggestion
**Note**: The `window.DD_LOGS` check prevents issues when a loading failure occurs with the SDK.
```

Author: The wording was used in 8 spots - all of them have been updated.
